src/compiler/transformers/module/module.ts

```diff
@@ -214,20 +214,28 @@ namespace ts {
         function addExportEqualsIfNeeded(statements: Statement[], emitAsReturn: boolean) {
             if (exportEquals && resolver.isValueAliasDeclaration(exportEquals)) {
                 if (emitAsReturn) {
-                    statements.push(createReturn(exportEquals.expression));
+                    const statement = createReturn(
+                        exportEquals.expression,
+                        /*location*/ exportEquals
+                    );
+
+                    setNodeEmitFlags(statement, NodeEmitFlags.NoTokenSourceMaps | NodeEmitFlags.NoComments);
+                    statements.push(statement);
                 }
                 else {
-                    statements.push(
-                        createStatement(
-                            createAssignment(
-                                createPropertyAccess(
-                                    createIdentifier("module"),
-                                    "exports"
-                                ),
-                                exportEquals.expression
-                            )
-                        )
+                    const statement = createStatement(
+                        createAssignment(
+                            createPropertyAccess(
+                                createIdentifier("module"),
+                                "exports"
+                            ),
+                            exportEquals.expression
+                        ),
+                        /*location*/ exportEquals
                     );
+
+                    setNodeEmitFlags(statement, NodeEmitFlags.NoComments);

```

Comment:
The `setNodeEmitFlags` function is bound to the current transformation (it is a member of `TransformationContext`) and isn't generally available to the create* functions in factory.ts.